### PR TITLE
fix(surveys): open pasted links in a new tab and keep element text selectable

### PR DESCRIPTION
**Pull request by jobenjada**
## What & why

Two respondent-facing bugs reported by a government customer.

**1. Links pasted into the editor open in the same tab.** The link toolbar sets `target="_blank" rel="noopener noreferrer"`, but a URL typed or pasted as plain text is turned into a link by the Lexical auto-link plugin, whose matcher set no attributes at all. Following such a link mid-survey navigated the respondent away from their in-progress response.

- `auto-link-plugin.tsx` — the URL matcher now returns the same attributes the link toolbar applies (`mailto:` matches deliberately keep the current tab).
- `sanitizeSurveyHtml()` in `packages/survey-ui` (re-exported from `packages/surveys`, so both renderers share one implementation) — the render path now sets `target="_blank" rel="noopener noreferrer"` on links that have no explicit target, so **surveys authored before this fix are fixed too**, without re-saving them. An explicit `target="_self"` is left alone. The DOMPurify hook is added and removed around each call, so no other sanitize call is affected.

**2. Question text can't be selected or copied.** Every `Label` variant in `survey-ui` carried `select-none`, and element headlines/descriptions render through it (`ElementHeader`). Respondents could not select the question text — to copy it, look it up, or paste it into a translator. Now only labels that act as controls (choice options, cards) keep `select-none`; `headline` and `description` are `select-text`. The class list was also collapsed into one `baseClasses` const instead of being repeated across four branches.

## Linear ticket

No ticket — filed directly from a customer-feedback review. Happy to link one if you'd rather track it.

## How this was tested

- `packages/surveys`: `npx vitest run` → 26 files, 666 tests ✅
- `packages/survey-ui`: `npx vitest run --coverage` → 5 files, 90 tests ✅ (new `utils-sanitize.test.ts`; the new lines are fully covered — the remaining gap in `utils.ts` is the pre-existing `getRTLScaleOptionClasses`)
- `apps/web`: `npx vitest run modules/ui/components/editor` → 11 tests ✅
- `pnpm build --filter=@formbricks/surveys... --force` ✅ (typecheck + bundle)
- `eslint` on every changed file → 0 errors, `pnpm format` clean
- Text selection verified in Chrome on a reduced repro of the label markup: with `user-select: none` a drag over the label selects nothing and focuses the input; with `user-select: text` the text selects and focus is *not* stolen (Chrome skips label activation when the click was a drag-select), so clicking the headline still focuses the input as before.

New tests cover: link without target gets `_blank` + `rel`, existing `_blank` keeps `rel`, explicit `_self` untouched, scripts/inline styles/event handlers still stripped, empty input, and that the hook does not leak into later sanitize calls.

## Breaking changes

None

## QA / Test Plan

**How to test**

- [ ] Editor → open a survey, in a question headline **paste or type a bare URL** (e.g. `https://formbricks.com`) so it auto-links → publish → open the survey link → click the link → it opens in a **new tab** and the survey stays open with answers intact.
- [ ] Same check for a link added with the toolbar's link button (unchanged behavior) → opens in a new tab.
- [ ] Type an email address as plain text (e.g. `hi@formbricks.com`) → it auto-links → clicking it opens the mail client, **not** a blank tab.
- [ ] Existing survey (created before this build) with a pasted URL in a headline or description → clicking it now also opens a new tab, without editing or re-saving the survey.
- [ ] Live survey → drag across a **question headline** and its description → the text highlights and Cmd/Ctrl+C copies it.
- [ ] Open text question → click (don't drag) the headline → the input is still focused, as before.
- [ ] Single/multi select → drag across an **answer option** label → it does not highlight, and clicking the option still selects it.
- [ ] Rich-text headline containing bold/italic and a link → renders correctly, link opens in a new tab, and the text is still selectable.

**Preconditions / test data**

- Any workspace; one link survey with a question whose headline contains a pasted URL, and one with a toolbar-inserted link.
- For the "existing survey" step, use a survey created before this build (or one whose headline HTML has no `target` attribute).

**Risks & regressions**

- Sanitization of survey content (headline, description, labels) — check that rich text, recall values and RTL content still render, and that no styles/scripts leak through.
- Label click-to-focus association for open text, consent, and choice questions.
- Survey preview inside the editor uses the same renderer — re-check there as well as in a live link survey.

**Migrations / env / cutover**

- none

### CODE REVIEW

File: apps/web/modules/ui/components/editor/components/auto-link-plugin.tsx [hunk 1]
```diff
@@ -4,6 +4,15 @@ const URL_MATCHER =
   /((https?:\/\/(www\.)?)|(www\.))[-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_+.~#?&//=]*)/;
 
 const EMAIL_MATCHER = /\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,}\b/;
+
+// Auto-linked URLs must behave like links inserted through the link toolbar, which
+// sets the same attributes: a survey opened in the same tab navigates the respondent
+// away from an in-progress response.
+const LINK_ATTRIBUTES = {
+  target: "_blank",
+  rel: "noopener noreferrer",
```
[the remark this answers is not in the document]
Comment (itsjavi) [reply apps/web/modules/ui/components/editor/components/auto-link-plugin.tsx]: Fixed in `8004d85`, with a follow-up in `14d3808`.

You called it correctly that `AGENTS.md` rules out a `.tsx` unit test here — so rather than reach for Playwright or accept it as manual-QA-only, the matchers moved to where a unit test is the *right* tool. `MATCHERS`, `LINK_ATTRIBUTES` and the two regexes now live in a plain `auto-link-matchers.ts`; `auto-link-plugin.tsx` is down to the four lines that render the plugin. The matchers were always pure functions, so this lands them on the "pure logic or edge cases: unit test" side of the AGENTS.md heuristic instead of the component side.

`auto-link-matchers.test.ts` covers 7 cases: the URL matcher returns `target="_blank" rel="noopener noreferrer"` for `https://`, `http://` and `www.` forms; index/length are correct for a URL embedded in surrounding text; and the email matcher returns `mailto:` with **no** attributes, pinning the deliberate same-tab behaviour you'd otherwise have to infer from the absence of a line. Reverting the attributes now fails CI.

Two things worth flagging rather than burying:

- The matchers were typed `(text: any)`. They now use Lexical's own `LinkMatcher`, so the return shape is checked against `LinkMatcherResult`/`AutoLinkAttributes` at compile time — a typo in `target` or `rel` is a type error, not a silent no-op.
- Moving the regex line made it "new code" to Sonar, which then flagged `S5869`: the URL matcher's character class listed `/` twice (`&//=`). Pre-existing, harmless, but real — deduped in `14d3808` after diffing old against new across 7 URL shapes to confirm identical match text and offset. A repeated literal inside a character class is a no-op.

Same commit also cleared the `S3776` cognitive-complexity finding on `Label` (16 → 12): the `variant` if/else chain became a `VARIANT_CLASSES` lookup, with `select-none`/`select-text` folded into it so each variant maps to one complete class string. Rendered `className` ordering is unchanged.

Verified: `survey-ui` 90 ✅, `surveys` 666 ✅, editor 18 ✅ (7 new), and all 16 checks green on `14d3808` with SonarCloud back to 0 open issues.

One gap I did **not** close: the PR body's "How this was tested" section doesn't mention `auto-link-matchers.test.ts`, since I didn't want to rewrite @jobenjada's prose. Worth a one-line update before merge.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by itsjavi — APPROVED** [review]
thanks I applied some fixes and LGTM
